File: modules/agent/src/schemas/project.py
"""Project schemas for API validation."""

from typing import Optional

from pydantic import BaseModel


class ProjectCreate(BaseModel):
    name: str
    description: Optional[str] = None
    status: Optional[str] = None
    current_status_id: Optional[int] = None
    start_date: Optional[str] = None
    end_date: Optional[str] = None


class ProjectUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    status: Optional[str] = None
    current_status_id: Optional[int] = None
    start_date: Optional[str] = None
    end_date: Optional[str] = None

File: modules/agent-go/internal/schemas/project.go
package schemas

// ProjectCreate represents the request body for creating a project
type ProjectCreate struct {
	Name            string  `json:"name" validate:"required"`
	Description     *string `json:"description"`
	Status          *string `json:"status"`
	CurrentStatusID *int64  `json:"current_status_id"`
	StartDate       *string `json:"start_date"`
	EndDate         *string `json:"end_date"`
}

// ProjectUpdate represents the request body for updating a project
type ProjectUpdate struct {
	Name            *string `json:"name"`
	Description     *string `json:"description"`
	Status          *string `json:"status"`
	CurrentStatusID *int64  `json:"current_status_id"`
	StartDate       *string `json:"start_date"`
	EndDate         *string `json:"end_date"`
}
